### PR TITLE
Aries-dev

### PR DESCRIPTION
**Pull request by asa1997**
- Added the environment script and test script for Aries-dev environment
- Updated code for the commands

### CODE REVIEW

**Comment by asa1997** [comment]
@EtricKombat  Please test this out by raising a pull request from https://github.com/asa1997/KOBman/tree/aries.
- Do make a release before you try it out. 
- Try out all the commands as well.
- Install command,
    - The changes are made in the validating version format function. It can now accept version numbers with four sets as well as 
1.0.0-alpha < 1.0.0-alpha.1 < 1.0.0-alpha.beta < 1.0.0-beta < 1.0.0-beta.2 < 1.0.0-beta.11 < 1.0.0-rc.1 < 1.0.0.
- Uninstall command,
   - Code updated for processing uninstall command with `all` parameter.
- List command,
   - A comma is added between the version numbers and a hyphen between a env name and the version numbers.
- Status command,
   - Earlier it could not process the env name Aries-dev, due to the hyphen in between. It has been rectified.
- Remove command,
   - A bit of code refactoring has also been done.

**Comment by asa1997** [comment]
Steps for running the test script for Aries-dev.
1. Install the environment using install command.
2. Move into the test folder and execute the below command,
    `./test-kob-Aries-dev.sh`

Output
-----------
- Success
Screenshot from 2020-08-19 17-23-51

- Failing condition, where Aries-dev was not installed prior to the execution of test script.
Screenshot from 2020-08-19 17-22-34

**Comment by asa1997** [comment]
@panickervinod @sumodgeorge @EtricKombat This pull request resolves issue #155 #153

**Comment by EtricKombat-zz** [comment]
Hi @asa1997 , while trying to install Aries_dev hitting out with minor errors. 
1) Aries installation issue
2) in the local system (.kobman) aries file residing in wrong location

Please find the issue ( https://github.com/asa1997/KOBman/issues/33 ) raised for the above mentioned error

**Comment by EtricKombat-zz** [comment]
@asa1997  , i have made the modification as you instructed in . Now the above listed issue is not encountering.

Listing down the snapshot of test script output

image

**Comment by EtricKombat-zz** [comment]
@sumodgeorge @panickervinod , this has been reviewed and tested out . LGTM

File: src/main/bash/commands/kobman-install.sh [hunk 1]
```diff
@@ -21,11 +21,10 @@ function __kobman_validate_environment
 }
 
 function __kobman_validate_version_format
```
Comment (panickervinod) [inline src/main/bash/commands/kobman-install.sh:23]: @asa1997 Can this function be moved to environment helper?

File: dist/environments [hunk 2]
```diff
@@ -1 +1 @@
-KOBman
+KOBman,Aries-dev
```
Comment (panickervinod) [inline dist/environments:1]: @asa1997  is this file required in every environment repo?

File: src/main/bash/commands/kobman-remove.sh [hunk 3]
```diff
@@ -4,17 +4,25 @@ function __kob_remove
...
-    if [[ -d $KOBMAN_ENV_ROOT ]]; then
-        __kobman_echo_no_colour "Removing dev environment"
-        rm -rf $KOBMAN_ENV_ROOT
-    fi
-
+    # if [[ -d $KOBMAN_ENV_ROOT ]]; then
+    #     __kobman_echo_no_colour "Removing dev environment"
+    #     rm -rf $KOBMAN_ENV_ROOT
+    # fi
+    __kobman_secure_curl "https://raw.githubusercontent.com/$KOBMAN_NAMESPACE/KOBman/master/dist/environments" >> $HOME/env_tmp.txt
```
Comment (panickervinod) [inline src/main/bash/commands/kobman-remove.sh:11]: @asa1997  Name the tmp file with what variable or value will be saved in it. Standardized naming for all temp file with a tmp prefix

File: src/main/bash/envs/kobman-Aries-dev.sh [hunk 4]
```diff
@@ -0,0 +1,176 @@
+#!/bin/bash
+function __kobman_install_Aries-dev
```
Comment (panickervinod) [inline src/main/bash/envs/kobman-Aries-dev.sh:2]: @asa1997 is the A cap consistent with all the environment names, let us have all function names small caps

File: src/main/bash/envs/kobman-Aries-dev.sh [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+	# if [[ "$?" == "0" ]]; then
+	# 	__kobman_echo_red "You have unsaved works"
+	# 	__kobman_echo_red "Uninstalling will remove all of the work done"
+	# 	__kobman_interactive_uninstall || return 1
+		# rm -rf $KOBMAN_ARIES_ENV_ROOT
+	# else
+	# 	rm -rf $KOBMAN_ARIES_ENV_ROOT
+	# fi
+}
+function __kobman_check_python3_installation
```
Comment (panickervinod) [inline src/main/bash/envs/kobman-Aries-dev.sh:91]: @asa1997 __kobman_check_python need not be an __kobman core function. it can be a core function if you refactor this out a common or a library.

File: src/main/bash/envs/kobman-Aries-dev.sh [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+{
+    __kobman_echo_no_colour "Checking python installation"
+    if [[ -z $(which python3) ]]; then
+        __kobman_echo_no_colour "Installing python3"
+        sudo apt-get update
+        sudo apt-get install python3.6
+    fi
+}
+function __kobman_check_pip_installation
```
Comment (panickervinod) [inline src/main/bash/envs/kobman-Aries-dev.sh:100]: __kobman_check_xxx need not be an __kobman core function. it can be a core function if you refactor this out a common or a library

**Review by panickervinod — COMMENTED** [review]
@asa1997  Please address the comments.

**Comment by panickervinod** [comment]
@asa1997 Please do address the comments.

File: src/main/bash/commands/kobman-install.sh (hunk 1 above)
Comment (asa1997) [reply src/main/bash/commands/kobman-install.sh:23]: It has been moved to env helper along with other validation functions

File: dist/environments (hunk 2 above)
Comment (asa1997) [reply dist/environments:1]: Yes, for now. This file is used to move  the environment files from stage/ folder to envs/ folder during the installation of the utility

File: src/main/bash/commands/kobman-remove.sh (hunk 3 above)
Comment (asa1997) [reply src/main/bash/commands/kobman-remove.sh:11]: File name has been changed

File: src/main/bash/envs/kobman-Aries-dev.sh (hunk 4 above)
Comment (asa1997) [reply src/main/bash/envs/kobman-Aries-dev.sh:2]: kob install -env Aries-dev
- We have to follow the name of the env given in the installation command as such.
- Aries-dev from install command is stored to a vairable and using this variable install function of the environment is called.
 $environment=Aries-dev
 __kobman_install_$environment "environment"
 - The first $environment is used to identify the installation function of the environment.

File: src/main/bash/envs/kobman-Aries-dev.sh (hunk 5 above)
Comment (asa1997) [reply src/main/bash/envs/kobman-Aries-dev.sh:91]: Function has been renamed
